app/app/src/main/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenter.java

```diff
@@ -34,17 +35,50 @@ public void onClearButtonPress() {
         view.handleOperations(false);
     }
 
+    public void onDeleteButtonPress() {
+        String reduction = EMPTY_STRING;
+
+        if (model.getOperator().isEmpty()) {
+            reduction = model.getFirstNumber();
+            if (!reduction.isEmpty()) {
+                reduction = reduction.substring(0, reduction.length() - 1);
+                model.setFirstNumber(reduction);
+                view.showNumberPressed(model.getFirstNumber());
+                controlDot(model.getFirstNumber());
+            }
+        } else {
+            reduction = model.getSecondNumber();
+            if (!reduction.isEmpty()) {
+                reduction = reduction.substring(0, reduction.length() - 1);
+                model.setSecondNumber(reduction);
+                view.showNumberPressed(model.getSecondNumber());
+                controlDot(model.getSecondNumber());
+            }
+        }
+    }
+
     public void onOperatorPressed(String symbol) {
-        model.setOperator(symbol);
-        view.showOperationPressed(symbol);
-        view.handleDot(true);
+        if (model.getOperator().isEmpty() || model.getSecondNumber().isEmpty()) {
+            model.setOperator(symbol);
+            view.showOperationPressed(symbol);
+            view.handleDot(true);
+        } else if (!model.getSecondNumber().isEmpty()) {
+            model.setFirstNumber(calculateResult());
+            view.showResult(model.getFirstNumber());
+            model.setSecondNumber(EMPTY_STRING);
+            model.setOperator(symbol);
+            view.showOperationPressed(symbol);
+            view.handleDot(true);
```

Comment:
this code is repeated at line 62, is there any way to solve this?

Reply:
i extract those duplicated lines into a new method that solve them